pkg/agent/route/route_windows.go

```diff
@@ -174,6 +186,107 @@ func (c *Client) DeleteRoutes(podCIDR *net.IPNet) error {
 	return nil
 }
 
+func (c *Client) AddVirtualIPToAntreaGW0Routes() {
+	metric := util.MetricHigh
+	linkIndex := c.nodeConfig.GatewayConfig.LinkIndex
+	gwIP := c.nodeConfig.GatewayConfig.IPv4
+
+	// A virtual IP is used to avoid too many ARP responders in OVS flows.
+	// Route: Virtual IP (169.254.169.252) -> antrea-gw0
+	vRoute := &util.Route{
+		LinkIndex:         linkIndex,
+		DestinationSubnet: &virtualGWIPNet,
+		GatewayAddress:    gwIP,
+		RouteMetric:       metric,
+	}
+	util.NewNetRoute(vRoute)
+
+	// For ClusterIP Service from host, reply traffic needs a route entry to route packet to vIP via antrea-gw0.
+	// If the NextHop of a route is antrea-gw0, then set it as 0.0.0.0. As a result, on-link/0.0.0.0 is in PersistentStore.
+	repRoute := &util.Route{
+		LinkIndex:         linkIndex,
+		DestinationSubnet: &net.IPNet{IP: config.ServiceGWHairpinIPv4, Mask: net.CIDRMask(32, 32)},
+		GatewayAddress:    net.IPv4zero,
+		RouteMetric:       metric,
+	}
+	util.NewNetRoute(repRoute)
+}
+
+func (c *Client) AddServiceRoutes(svcIP net.IP, gwIP net.IP) error {
```

Comment:
Is it possible to share the same code with Linux on this function?

Reply:
Used the same function signature.